### DISCUSSION TITLE
Using flat config from the CLI / ESLint class

### DISCUSSION THREAD

**Discussion by tunnckoCore** [eslint/eslint#16079] [General] [answered]
Heya there.

Is there a way to force the use of flat config through the CLI, or when using the ESLint class programmatically?

I'm currently playing with the Linter, but it's.. as everything in ESLint, partially accepting one thing, and partially doing one thing. Thankfully there is this ability to pass `configType` which saves couple of seconds run time.

Or what's the way to pass full resolved config to Linter? I'm digging into whole eslint source for days, so.. I see that you can use `defineRule(s)` and `defineParser`, but seems that passing `languageOptions.globals` isn't fully respected (yet?). Passing `languageOptions.globals: { console: true }` is accepted, but when pass `buffer: true` or `Buffer: true` it isn't, but it's reported as `no-undef`.

**Comment by nzakas** [comment, marked as answer]
Flat config is not yet supported and should not be used. What’s checked in is just the beginning of support. 

> Or what's the way to pass full resolved config to Linter?

I’m not sure what you’re asking here. Can you provide more details?
